tests/Jira/ApiTest.php

```diff
@@ -14,116 +15,164 @@
-	public function testSetEndpointTrailingSlash()
+	protected $credential;
+
+	/**
+	 * Client.
+	 *
+	 * @var ObjectProphecy
+	 */
+	protected $client;
+
+	protected function setUp()
 	{
-		$api = new Api('https://test.test/', new Anonymous(), null);
-		$this->assertEquals('https://test.test', $api->getEndpoint());
+		parent::setUp();
 
-		// Make sure nothing is removed if there is no trailing slash
-		$url = 'https://urlwithouttrailing.slash';
-		$api->setEndPoint($url);
-		$this->assertEquals($url, $api->getEndpoint());
+		$this->credential = $this->prophesize('chobie\Jira\Api\Authentication\AuthenticationInterface')->reveal();
+		$this->client = $this->prophesize('chobie\Jira\Api\Client\ClientInterface');
+
+		$this->api = new Api(self::ENDPOINT, $this->credential, $this->client->reveal());
 	}
 
 	/**
-	 * Tests that the updateVersion call constructs the correct api call
+	 * @dataProvider setEndpointDataProvider
 	 */
+	public function testSetEndpoint($given_endpoint, $used_endpoint)
+	{
+		$api = new Api($given_endpoint, $this->credential, $this->client->reveal());
+		$this->assertEquals($used_endpoint, $api->getEndpoint());
+	}
+
+	public function setEndpointDataProvider()
+	{
+		return array(
+			'trailing slash removed' => array('https://test.test/', 'https://test.test'),
+			'nothing removed' => array('https://test.test', 'https://test.test'),
+		);
+	}
+
 	public function testUpdateVersion()
 	{
 		$params = array(
-			'released' => true,
-			'releaseDate' => '2010-07-06',
+			'overdue' => true,
+			'description' => 'new description',
 		);
 
-		// Stub the api method and keep the rest intact
-		/** @var Api|\PHPUnit_Framework_MockObject_MockObject $api */
-		$api = $this->getMockBuilder('\chobie\Jira\Api')->setMethods(array('api'))->disableOriginalConstructor()->getMock();
-		$api->expects($this->once())->method('api')->with(
-			$this->equalTo(Api::REQUEST_PUT),
-			$this->equalTo('/rest/api/2/version/111000'),
-			$this->equalTo($params)
+		$this->expectClientCall(
+			Api::REQUEST_PUT,
+			'/rest/api/2/version/111000',
+			$params,
+			''
 		);
 
-		$api->updateVersion(111000, $params);
+		$this->assertFalse($this->api->updateVersion(111000, $params));
 	}
 
-	/**
-	 * Tests that the releaseVersion call constructs the correct api call
-	 */
-	public function testReleaseVersion()
+	public function testReleaseVersionAutomaticReleaseDate()
 	{
 		$params = array(
 			'released' => true,
 			'releaseDate' => date('Y-m-d'),
 		);
 
-		// Stub the api method and keep the rest intact
-		/** @var Api|\PHPUnit_Framework_MockObject_MockObject $api */
-		$api = $this->getMockBuilder('\chobie\Jira\Api')->setMethods(array('api'))->disableOriginalConstructor()->getMock();
-		$api->expects($this->once())->method('api')->with(
-			$this->equalTo(Api::REQUEST_PUT),
-			$this->equalTo('/rest/api/2/version/111000'),
-			$this->equalTo($params)
+		$this->expectClientCall(
+			Api::REQUEST_PUT,
+			'/rest/api/2/version/111000',
+			$params,
+			''
```

Comment:
Is this comment addressed to the right line of code?

Ther `''` you've selected for comment is what `ClientInterface` implementing class returns. And it always returns strings, because that's what HTTP response is. It's `Api::api` that is json-decoding them and wrapping them into `Result` class.